### DISCUSSION TITLE
Elastalert scan all the documents in new timeframe?

### DISCUSSION THREAD

**Discussion by ishukeshri2712** [jertel/elastalert2#1502] [Q&A] [answered]
@jertel I have another doubt suppose we have an elastalert rule which query over all documents and we have put timestamp of last 5 minutes then it will query fresh documents which are pushed to elasticsearch ?

**Comment by jertel** [comment, marked as answer]
Are you asking if ElastAlert 2 will find a backdated log that was ingested with a timestamp value earlier than the last ElastAlert 2 query? If so then that depends on the rule configuration. I suggest looking at elastalert.py::set_starttime() to see all nuances involved.
